### PR TITLE
Preparation for basops integration with 3GPP update

### PR DESCRIPTION
**Pull request by ludomal**
I have restored some of the code formatting from the original STL2009 for files where indentation helps for the understanding of the code.

### CODE REVIEW

File: src/basop/test_framework/tools/csv2xl.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import os
```
Comment (dennisguse) [inline src/basop/test_framework/tools/csv2xl.py]: Why should this file be required?
It converts CSV to Excel, or?

File: src/basop/test_framework/tools/array_1000.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import random;
```
Comment (dennisguse) [inline src/basop/test_framework/tools/array_1000.py]: What does this program do?

File: src/basop/test_framework/tools/arrange_data_n_plots.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import xlrd
```
Comment (dennisguse) [inline src/basop/test_framework/tools/arrange_data_n_plots.py]: What does this program do?

File: src/basop/test_framework/test/test_precision.c [hunk 4]
```diff
@@ -41,6 +35,14 @@ FILE *fp_spcl;
          }                                           \
 }
 
+// 2 input 1 output
+#define FUNC_TEST_2ISO_SHIFT_REF(func_ptr, in1, in2, out, loop_cnt){ \
```
Comment (dennisguse) [inline src/basop/test_framework/test/test_precision.c]: This looks very ugly.
What is the reason for the use of a define here?

**Comment by ahomkar** [comment]
@dennisguse
The STL BASOP test framework has two modes

  1.  Sanity test : to verify basic sanity that package works on given platform.
  2.  Full regression :  Full regression  to test all STL17 BASOP.  It dumps the DUT and REF o/p , difference and some other params as csv files.   It also prints the results summary .

All the .py scripts are for the expert user which help to convert these .csv files to various plots (inside xls).



Ludovic has given us suggestion to document the various .py scripts.  We will add that info in next couple of days.
Thanks,
   Ajay

**Comment by dennisguse** [comment]
@ahomkar I am concerned about the maintenance and technical debt.
In some years somebody will have to clean them up.
And at the moment, there is a high number of things that need cleaning - why add one more?

In addition, it adds a another programming language - so far STL was mainly C/CPP.
The actual question is: how many dependencies need to be survive the years, so the code remains useful?

Moreover, there are so many possible ways to plot a graph.
Why chose Microsoft Excel?
I would suggest to not include these scripts, so everybody can do the evaluation using his/her own tools.

**Comment by ludomal** [comment]
Thanks Dennis for the comments. I have forgotten that this branch of my repository was still linked to a pull request. I am still in the middle of integrating the new package that I received. I suggest we close this pull request and open a new one once the code is ready.

Regarding the Python code, they still need to be sanitized, and I agree this adds a new langage and more things to tidy up/maintain. I do see values in these two scripts as they provide examples:
* array_1000.py is a helper tool that generates sets of random numbers as test vectors. It produces custom_data.h.
* plot_graphs.py interprets the csv files generated by the new test framework and produces plots (using matplotlib).

I do not think csv2xl.py and arrange_data_n_tools.py are necessary.

**Pull request closed without merging**